### ISSUE TITLE
Examples will not compile on Windows

### ISSUE THREAD

**Issue by AnonMonkey** [hal7df/iash#11]
**Making the example program on Windows causes this error, stating that "iash.h" couldn't be found:**
```make
examples/basic.cpp:24:10: fatal error: iash.h: No such file or directory
 #include "iash.h"
          ^~~~~~~~
compilation terminated.
make: *** [bin/example/basic.static] Error 1
```

**Comment by hal7df** [comment]
Oddly, an unmodified Makefile will correctly build the project under [MSYS2](http://www.msys2.org/). Can you check to see if the most recent commit (de8a285) works under the older MSYS?

**Comment by hal7df** [comment]
Closing this as compilation works reliably under MSYS2.

**Comment by hal7df** [comment]
Well, looks like the msys2 build is still suboptimal...won't run outside of msys2. Will investigate further.

**Comment by AnonMonkey** [comment]
In addition to this, building with msys2 compiles using the __unix portions of the code instead of __WIN32.

**Comment by hal7df** [comment]
Hmm. Maybe msys2 isn't the right way to go about this.

Another thought would be to get visual studio c++ and write a new makefile but that would be a lot of effort. Best way forward is probably with the original MinGW (since I know for sure it was using the Win32 API). Still not sure why it doesn't like the `-I` flag...

**Comment by AnonMonkey** [comment]
Tried building with MinGW, comes back with this error:
``` make
The syntax of the command is incorrect.
mingw32-make: *** [Makefile:62: bin/obj/CommandDispatcher.o] Error 1
```
I'm not sure if this is an issue with the command or with the makefile.

**Comment by AnonMonkey** [comment]
As a follow up, I think these two lines are causing the error:
```make
41 MKDIR = mkdir -p
62 @$(MKDIR) $(dir $@)
```

According to the [Microsoft docs](https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/md), there is no support for the -p option, where in the [Linux man pages](https://linux.die.net/man/1/mkdir) there is. This discrepancy is probably what's causing the syntax error.

If this is the case, we should review the makefile and generate a new one if necessary to support both Windows and Linux.

**Comment by hal7df** [comment]
So the `-p` switch serves two functions: to make multiple nested directories in one call, and fail silently if the directory(ies) already exist. If you can create a similar call for `mkdir` in Windows, then we can make the makefile change `$(MKDIR)` depending on the OS.

**Comment by hal7df** [comment]
Possible fix in 870ca80, will investigate when my Windows VM decides to behave.

**Comment by hal7df** [comment]
The two latest commits I pushed have iash building in wine (Win10 compatibility mode) with no issues. Since it's a console app, it should be working in native Windows as well. Some other thoughts:

- MinGW doesn't link one of the `cstdio` functions properly for some reason, so if you get an error like `multiple definition of vsnprintf`, go into mingw-get and install the `mingw32-libmingwex-dev` and `mingw32-libmingwex-dll` packages (more information [here](https://stackoverflow.com/q/43294488))
- I rewrote the Makefile to use as few Unix tools as possible on Windows, so running `mingw32-make static-example` from the root of the repository in cmd should do the trick. MSYS bash/make probably won't like it.
